src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -221,15 +245,18 @@ private string GetCommitsWorkFolder(IExecutionContext context)
 
                 await retryExecutor.ExecuteAsync(
                     async () =>
-                        {
-                            var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
-                            executionContext.Output(StringUtil.Loc("RMEnsureArtifactFolderExistsAndIsClean", downloadFolderPath));
-                            releaseFileSystemManager.EnsureEmptyDirectory(downloadFolderPath, executionContext.CancellationToken);
+                    {
+                        var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
+                        executionContext.Output(StringUtil.Loc("RMEnsureArtifactFolderExistsAndIsClean",
+                            downloadFolderPath));
+                        releaseFileSystemManager.EnsureEmptyDirectory(downloadFolderPath,
```

Comment:
shouldn't this happen in Initialize with this refactoring? the download artifact tasks would expect the folder to be empty, so agent has to empty it in initialize itself as download as part of agent might not happen?

Reply:
added CreateArtifactsFolder in initialize, if no artifacts to be downloaded by agent.